libpod/container_api.go

```diff
@@ -175,9 +173,27 @@ func (c *Container) StartAndAttach(ctx context.Context, streams *define.AttachSt
 		}
 	}
 
-	if err := c.prepareToStart(ctx, recursive); err != nil {
-		return nil, err
+	if c.state.State != define.ContainerStateRunning {
+		if !start {
+			return nil, errors.New("you can only attach to running containers")
+		}
+
+		if err := c.prepareToStart(ctx, true); err != nil {
+			return nil, err
+		}
+	}
+
+	if !start {
```

Comment:
Why only !start on these checks? I can't imagine starting and attaching to a PassthroughLogging container makes sense

Reply:
Is the comment below not good enough? This code path is used by podman run so we cannot reject it on a start. We are not doing a real attach, you can see the short cut in the ConmonOCIRuntime Attach interface

Reply:
It feels fundamentally incorrect to be calling the Attach API when we have no intention of actually attaching but it seems like the fix would be very intrusive (having to look up log driver in the frontend code, and spreading the concern of log driver into the frontend as well). I suppose we could do an ErrCannotAttach for these log drivers and retry with the standard Start API in the frontend code if we get that, but that seems sufficiently separate to not block this.

Reply:
Normal start would work but it is more communicated in the fronted because we still have to call Wait() in order to wait for the container to exit which is why this code path makes most sense currently.

But yeah I agree it is confusing and does not make much sense logically within Attach()